### ISSUE TITLE
Ubuntu 16.04, SteamCMD network failure

### ISSUE THREAD

**Issue by Cybroken** [docker/docker#22304]
Hey guys, I got a problem setting up SteamCMD correctly. Ubuntu 15.10 on my hostsystem was working, but now Ubuntu 16.04 does not work anymore. It says, that SteamCMD could not establish a connection to steam. If I execute the whole thing on my hostmachine directly, it works perfectly.

I did not modify the Dockerfile or any other files during update to 16.04. 
I tried docker version 1.11 and newest 1.12.0-dev.

**docker-compose.yml**
```
csgo_server:
  build: ${DOCKER_IMAGES}/csgo/
  ports:
    - "27015:27015/udp"
    - "27015:27015/tcp"
```

**Dockerfile**
```
# Dockerfile to run a CS:GO server
FROM ubuntu:14.04

# Install needed files for SteamCMD + common tools
USER root
RUN dpkg --add-architecture i386
RUN apt-get update
RUN apt-get install -y libc6:i386 libstdc++6:i386 wget software-properties-common vim unzip
RUN useradd -ms /bin/bash csgo

# Copy dotfiles
RUN cd && cp -R .bashrc .profile /home/csgo

RUN chown -R csgo:csgo /home/csgo/

USER csgo
WORKDIR /home/csgo

# Download and unarchive SteamCMD
RUN wget https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz
RUN tar -xvzf steamcmd_linux.tar.gz

# Create directory for CS:GO
RUN mkdir strike
RUN chown -R csgo:csgo /home/csgo/strike/
RUN chmod -R 755 /home/csgo/strike/

# Server config
EXPOSE 27015

# Add server start script
ADD start.sh /home/csgo/start.sh

USER root
RUN chown csgo:csgo /home/csgo/start.sh
RUN chmod 755 /home/csgo/start.sh

# Run as non privileged user
CMD ["/home/csgo/start.sh"]
```

**start.sh:**

```
#!/bin/bash
./steamcmd.sh +login anonymous +quit
```

**Output of `docker version`:**

```
Client:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.5.4
 Git commit:   8a0d2d8-unsupported
 Built:        Sat Apr 23 10:17:24 2016
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.5.4
 Git commit:   8a0d2d8-unsupported
 Built:        Sat Apr 23 10:17:24 2016
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 12
 Running: 7
 Paused: 0
 Stopped: 5
Images: 83
Server Version: 1.12.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 184
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.0-21-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 5.828 GiB
Name: zero-division
ID: JWAQ:DCSW:5OJP:RLKL:UW47:XTFX:JWDP:4I6F:VCLW:LWMQ:7ULL:CZUI
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: true
Insecure registries:
 127.0.0.0/8
```

**Steps to reproduce the issue:**
Just execute the docker-compose file with given Dockerfile and start.sh on Ubuntu 16.04 (host)

**Describe the results you received:**
Console:
```
csgo_server_1 | ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
csgo_server_1 | [  0%] Checking for available update...
csgo_server_1 | SteamUpdater: Error: Download failed: http error 0
csgo_server_1 | [  0%] Download Complete.
csgo_server_1 | [----] Verifying installation...
csgo_server_1 | [  0%] Downloading Update...
csgo_server_1 | [  0%] Checking for available update...
csgo_server_1 | SteamUpdater: Error: Download failed: http error 0
csgo_server_1 | [  0%] Download Complete.
csgo_server_1 | SteamUpdater: Error: Steam needs to be online to update.  Please confirm your network connection and try again.
```

**Additional information you deem important (e.g. issue happens only occasionally):**
- Host: Ubuntu 16.04 x64

**Comment by thaJeztah** [comment]
Were you already running docker 1.11 before upgrading to ubuntu 16.04?

**Comment by Cybroken** [comment]
Yep, 1.11 was running perfectly without problems on 15.10. My actual container setup contains one teamspeak-, one apache-, and one nginx-server. Those are perfectly working! 
But SteamCMD-container does not work anymore.

**Comment by justincormack** [comment]
Likely to be also the same as #22296 as Steam uses Wine. Does it work if you use `docker run --security-opt seccomp=unconfined`?

**Comment by Cybroken** [comment]
Yes, this command did the trick! It works! 
But why? What does this command do?

**Comment by thaJeztah** [comment]
that command disables the seccomp profile, see https://docs.docker.com/engine/security/seccomp/.

let me close this as a duplicate of https://github.com/docker/docker/issues/22296 and https://github.com/docker/docker/issues/20550.

**Comment by krailler** [comment]
I have same issue.. But i use ./steamcmd.sh  for build, and --security-opt does not exist.. :(

**Comment by thaJeztah** [comment]
@krailler `--security-opt` is an option for `docker run`, so has to be provided *before* the image you're running, not after, because then it's treated as an option that's passed to the images' entrypoint. So; 

    docker run --security-opt seccomp=unconfined <your-image>

**Comment by krailler** [comment]
@thaJeztah .. So .. I can not make a build of an image in ubuntu 16.04? I have to do it in another machine and upload?

I have "SteamUpdater: Error: Download failed: http error 0" error on RUN ./steamcmd.sh +exit, in build.. Only in 16.04.. If not, I will have to revert to a previous version of ubuntu.

**Comment by justincormack** [comment]
This should be fixed in 1.11.2 and 1.12 via #22922
